image-hd.c

```diff
@@ -600,7 +600,7 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 				if (hd->gpt)
 					now = hd->gpt_location + (GPT_SECTORS - 1) * 512;
 			}
-			part->offset = roundup(now, hd->align);
+			part->offset = roundup(now, part->align ?: hd->align);
```

Comment:
Hmm, shouldn't this already fail above:
```
if (part->in_partition_table && (part->offset % hd->align)) {
```
I think `part->align` should be initialized with `hd->align` and then used unconditionally when appropriate.

Reply:
Perhaps, I'm not sure we know hd->align when we initialize part->align. But of course one can  "fixup" each part->align early in hdimage_setup(). In any case, please disregard this PR and the gpt overlap one, I'm going to submit a longer series, which will supersede both of those.